src/core/records-grant-authorization.ts

```diff
@@ -0,0 +1,47 @@
+import type { MessageStore } from '../types/message-store.js';
+import type { RecordsDelete } from '../interfaces/records-delete.js';
+import type { RecordsPermissionScope } from '../types/permissions-types.js';
+import type { RecordsRead } from '../interfaces/records-read.js';
+import type { RecordsWrite } from '../interfaces/records-write.js';
+
+import { DwnMethodName } from './message.js';
+import { GrantAuthorization } from './grant-authorization.js';
+import { DwnError, DwnErrorCode } from './dwn-error.js';
+
+export class RecordsGrantAuthorization {
+  /**
+   * Authorizes the scope of a PermissionsGrant for RecordsRead, RecordsWrite, and RecordsDelete messages.
+   */
+  public static async authorizeRecordsGrant(
+    tenant: string,
+    incomingMessage: RecordsRead | RecordsWrite | RecordsDelete,
+    messageStore: MessageStore,
+  ): Promise<void> {
+
+    // authorize generic message
+    const permissionsGrantMessage = await GrantAuthorization.authorizeGenericMessage(tenant, incomingMessage, messageStore);
+
+    const grantScope = permissionsGrantMessage.descriptor.scope as RecordsPermissionScope;
+
+    // if recordIds, authorizeRecordIds
+    if (grantScope.recordIds !== undefined) {
```

Comment:
It definitely seems useful to be able to grant access to multiple records, though we should get together tomorrow with @csuwildcat on the expectation around if a grant should align scoping to encryption key scope, which is one of: flat-space-schema, flat-space-schema-data-format, flat-space-data-format, protocol-context, protocol-path.

Reply:
I decided to repurpose this PR to support flat-space-schema instead of a list of record IDs. We can circle back to record IDs in a couple weeks when we have time to sync on it again.